src/myinfo-types.ts

```diff
@@ -1,160 +1,391 @@
 export enum MyInfoSource {
-    GovtVerified = '1',
-    UserProvided = '2',
-    NotApplicable = '3',
-    SingPassVerified = '4',
+  GovtVerified = '1',
+  UserProvided = '2',
+  NotApplicable = '3',
+  SingPassVerified = '4',
 }
 
-export enum MyInfoSex {
-    Male = 'M',
-    Female = 'F',
-    Unknown = 'U',
+interface IMetadata {
+  lastupdated: string
+  source: MyInfoSource
+  classification: 'C'
+  unavailable?: boolean
 }
 
-interface IMyInfoMetadata {
-    lastUpdated: string
-    source: MyInfoSource
-    classification: 'C'
+interface OptionalString {
+  value?: string
 }
 
-export interface IMyInfoBasicField extends IMyInfoMetadata {
-    value?: string
+interface OptionalNumber {
+  value?: number | ''
 }
 
-export interface IMyInfoHighLowField extends IMyInfoMetadata {
-    high: number
-    low: number
+interface OptionalBoolean {
+  value?: boolean | ''
 }
 
-export interface IMyInfoFieldWithDesc extends IMyInfoBasicField {
-    desc: string
+export interface IBasicField extends IMetadata, OptionalString {}
+
+interface CodeAndDesc {
+  code?: string
+  desc?: string
+}
```

Comment:
should tighten the types here as well - I'm pretty sure both `code` and `desc` have to appear together

Reply:
do we have a guarantee of that though? they seem to both be optional fields according to the Swagger docs. I tried to be conservative with the typing and not provide false guarantees, since you previously mentioned that the `value` field was sometimes missing.

Reply:
The specifications are supposed to guarantee it, under "Understanding the Data Structure (new in v3)":

> Each data property will contain either:
> 
> a value property, or
> a pair of code and desc properties, or
> an array of data properties, or
> other data properties

MyInfo v2 was dubious at respecting this - I found a couple of bugs myself - but MyInfo v3 should be different. Try testing against the Sandbox API instead of Mockpass? It's supposed to be a more authoritative source.